### ISSUE TITLE
[clangd] Assertion `!Ctx.getSourceManager().isLoadedSourceLocation(RC->getBeginLoc())` fail when background index enabled

### ISSUE THREAD

**Issue by hongtaihu** [llvm/llvm-project#198851]
When  background index enabled, the assertion is triggered; otherwise, it isn't triggered.

requests:
```
 initialize -> initialized -> didOpen -> shutdown -> exit
```

Source code(reduced from https://github.com/llvm/llvm-project/blob/main/clang/test/Modules/initializers.cpp):
```cxx
#pragma clang module build a
module a {}
#pragma clang module contents
#pragma clang module begin a
namespace b {
struct c {
  int d; //
};
} // namespace b
#pragma clang module end
#pragma clang module endbuild
import a
```

```
I[02:01:57.480] clangd version 23.0.0git (https://github.com/llvm/llvm-project.git c75e1e62d11be965a4dca0d816a4d19a6684b043)
I[02:01:57.480] Features: linux+debug-tidy
I[02:01:57.480] PID: 34
I[02:01:57.480] Working directory: <REPO_ROOT>/harness/cases/CASE-20260520-002/workspace
I[02:01:57.480] argv[0]: <CLANGD_BUILD>/bin/clangd
I[02:01:57.480] argv[1]: --clang-tidy=0
I[02:01:57.480] Starting LSP over stdin/stdout
I[02:01:57.488] <-- initialize(1)
I[02:01:57.491] --> reply:initialize(1) 2 ms
Content-Length: 2077

{"id":1,"jsonrpc":"2.0","result":{"capabilities":{"astProvider":true,"callHierarchyProvider":true,"clangdInlayHintsProvider":true,"codeActionProvider":true,"compilationDatabase":{"automaticReload":true},"completionProvider":{"resolveProvider":false,"triggerCharacters":[".","<",">",":","\"","/","*"]},"declarationProvider":true,"definitionProvider":true,"documentFormattingProvider":true,"documentHighlightProvider":true,"documentLinkProvider":{"resolveProvider":false},"documentOnTypeFormattingProvider":{"firstTriggerCharacter":"\n","moreTriggerCharacter":[]},"documentRangeFormattingProvider":{"rangesSupport":true},"documentSymbolProvider":true,"executeCommandProvider":{"commands":["clangd.applyFix","clangd.applyRename","clangd.applyTweak"]},"foldingRangeProvider":true,"hoverProvider":true,"implementationProvider":true,"inactiveRegionsProvider":true,"inlayHintProvider":true,"memoryUsageProvider":true,"referencesProvider":true,"renameProvider":true,"selectionRangeProvider":true,"semanticTokensProvider":{"full":{"delta":true},"legend":{"tokenModifiers":["declaration","definition","deprecated","deduced","readonly","static","abstract","virtual","dependentName","defaultLibrary","usedAsMutableReference","usedAsMutablePointer","constructorOrDestructor","userDefined","functionScope","classScope","fileScope","globalScope"],"tokenTypes":["variable","variable","parameter","function","method","function","property","variable","class","interface","enum","enumMember","type","type","unknown","namespace","typeParameter","concept","type","macro","modifier","operator","bracket","label","comment"]},"range":false},"signatureHelpProvider":{"triggerCharacters":["(",")","{","}","<",">",","]},"standardTypeHierarchyProvider":true,"textDocumentSync":{"change":2,"openClose":true,"save":true},"typeDefinitionProvider":true,"typeHierarchyProvider":true,"workspaceSymbolProvider":true},"serverInfo":{"name":"clangd","version":"clangd version 23.0.0git (https://github.com/llvm/llvm-project.git c75e1e62d11be965a4dca0d816a4d19a6684b043) linux+debug-tidy x86_64-unknown-linux-gnu"}}}I[02:01:57.500] <-- initialized
I[02:01:57.516] <-- textDocument/didOpen
I[02:01:57.517] Loaded compilation database from <REPO_ROOT>/harness/cases/CASE-20260520-002/workspace/compile_commands.json
I[02:01:57.517] ASTWorker building file <REPO_ROOT>/harness/cases/CASE-20260520-002/workspace/initializers.cpp version 1 with command
[<REPO_ROOT>/harness/cases/CASE-20260520-002/workspace]
/usr/lib/llvm-18/bin/clang++ --driver-mode=g++ -std=c++26 -fsyntax-only -I<REPO_ROOT>/harness/cases/CASE-20260520-002/workspace/Inputs -resource-dir=<CLANGD_BUILD>/lib/clang/23 -- <REPO_ROOT>/harness/cases/CASE-20260520-002/workspace/initializers.cpp
I[02:01:57.517] Enqueueing 1 commands for indexing
I[02:01:57.525] Built preamble of size 251424 for file <REPO_ROOT>/harness/cases/CASE-20260520-002/workspace/initializers.cpp version 1 in 0.01 seconds
I[02:01:57.526] Indexing c++23 standard library in the context of <REPO_ROOT>/harness/cases/CASE-20260520-002/workspace/initializers.cpp
clangd: <LLVM_PROJECT>/clang-tools-extra/clangd/CodeCompletionStrings.cpp:151: std::string clang::clangd::getDeclComment(const clang::ASTContext&, const clang::NamedDecl&): Assertion `!Ctx.getSourceManager().isLoadedSourceLocation(RC->getBeginLoc())' failed.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace and instructions to reproduce the bug.
Stack dump:
0.      <eof> parser at end of file
I[02:01:57.529] --> textDocument/publishDiagnostics
Content-Length: 1410

{"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"diagnostics":[{"code":"pp_module_build_missing_end","message":"No matching '#pragma clang module endbuild' for this '#pragma clang module build'","range":{"end":{"character":26,"line":0},"start":{"character":21,"line":0}},"severity":1,"source":"clang"},{"code":"pp_unexpected_tok_after_module_name","message":"Unexpected preprocessing token '{' after module name, only ';' and '[' (start of attribute specifier sequence) are allowed","range":{"end":{"character":10,"line":1},"start":{"character":9,"line":1}},"severity":1,"source":"clang"},{"code":"pp_unexpected_tok_after_module_name","message":"Unexpected preprocessing token '{' after module name, only ';' and '[' (start of attribute specifier sequence) are allowed","range":{"end":{"character":13,"line":2},"start":{"character":12,"line":2}},"severity":1,"source":"clang"},{"code":"pp_module_begin_wrong_module","message":"Must specify '-fmodule-name='m'' to enter submodule of this module","range":{"end":{"character":28,"line":5},"start":{"character":27,"line":5}},"severity":1,"source":"clang"},{"code":"module_not_found","message":"Module 'm' not found","range":{"end":{"character":8,"line":1},"start":{"character":7,"line":1}},"severity":1,"source":"clang"}],"uri":"file://<REPO_ROOT>/harness/cases/CASE-20260520-002/workspace/initializers.cpp","version":1}} #0 0x00005cfba86d58f2 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) <LLVM_PROJECT>/llvm/lib/Support/Unix/Signals.inc:884:3
 #1 0x00005cfba86d266c llvm::sys::RunSignalHandlers() <LLVM_PROJECT>/llvm/lib/Support/Signals.cpp:108:20
 #2 0x00005cfba86d2ee1 SignalHandler(int, siginfo_t*, void*) <LLVM_PROJECT>/llvm/lib/Support/Unix/Signals.inc:448:14
 #3 0x000071b0d5845330 (/lib/x86_64-linux-gnu/libc.so.6+0x45330)
 #4 0x000071b0d589eb2c __pthread_kill_implementation ./nptl/pthread_kill.c:44:76
 #5 0x000071b0d589eb2c __pthread_kill_internal ./nptl/pthread_kill.c:78:10
 #6 0x000071b0d589eb2c pthread_kill ./nptl/pthread_kill.c:89:10
 #7 0x000071b0d584527e raise ./signal/../sysdeps/posix/raise.c:27:6
 #8 0x000071b0d58288ff abort ./stdlib/abort.c:81:7
 #9 0x000071b0d582881b _nl_load_domain ./intl/loadmsgcat.c:1177:9
#10 0x000071b0d583b517 (/lib/x86_64-linux-gnu/libc.so.6+0x3b517)
#11 0x00005cfba9bb1dee decltype(auto) llvm::dyn_cast<clang::NamedDecl, clang::DeclContext const>(clang::DeclContext const*) <LLVM_PROJECT>/llvm/include/llvm/Support/Casting.h:656:3
#12 0x00005cfba9bb1dee decltype(auto) llvm::dyn_cast<clang::NamedDecl, clang::DeclContext const>(clang::DeclContext const*) <LLVM_PROJECT>/llvm/include/llvm/Support/Casting.h:655:37
#13 0x00005cfba9bb1dee clang::clangd::getDeclComment[abi:cxx11](clang::ASTContext const&, clang::NamedDecl const&) <LLVM_PROJECT>/clang-tools-extra/clangd/CodeCompletionStrings.cpp:118:46
#14 0x00005cfba9bb1f30 clang::clangd::getDocComment[abi:cxx11](clang::ASTContext const&, clang::CodeCompletionResult const&, bool) <LLVM_PROJECT>/clang-tools-extra/clangd/CodeCompletionStrings.cpp:97:36
#15 0x00005cfba9f50843 clang::clangd::SymbolCollector::addDeclaration(clang::NamedDecl const&, clang::clangd::SymbolID, bool) <LLVM_PROJECT>/clang-tools-extra/clangd/index/SymbolCollector.cpp:1119:60
#16 0x00005cfba9f51462 clang::clangd::SymbolCollector::handleDeclOccurrence(clang::Decl const*, unsigned int, llvm::ArrayRef<clang::index::SymbolRelation>, clang::SourceLocation, clang::index::IndexDataConsumer::ASTNodeInfo) <LLVM_PROJECT>/clang-tools-extra/clangd/index/SymbolCollector.cpp:728:33
#17 0x00005cfba9fa8d7b llvm::SmallVectorTemplateCommon<clang::index::SymbolRelation, void>::isSmall() const <LLVM_PROJECT>/llvm/include/llvm/ADT/SmallVector.h:148:39
#18 0x00005cfba9fa8d7b llvm::SmallVectorImpl<clang::index::SymbolRelation>::~SmallVectorImpl() <LLVM_PROJECT>/llvm/include/llvm/ADT/SmallVector.h:609:23
#19 0x00005cfba9fa8d7b llvm::SmallVector<clang::index::SymbolRelation, 6u>::~SmallVector() <LLVM_PROJECT>/llvm/include/llvm/ADT/SmallVector.h:1230:3
#20 0x00005cfba9fa8d7b clang::index::IndexingContext::handleDeclOccurrence(clang::Decl const*, clang::SourceLocation, bool, clang::Decl const*, unsigned int, llvm::ArrayRef<clang::index::SymbolRelation>, clang::Expr const*, clang::Decl const*, clang::DeclContext const*) <LLVM_PROJECT>/clang/lib/Index/IndexingContext.cpp:426:34
#21 0x00005cfba9fa93b6 clang::index::IndexingContext::handleDecl(clang::Decl const*, clang::SourceLocation, unsigned int, llvm::ArrayRef<clang::index::SymbolRelation>, clang::DeclContext const*) <LLVM_PROJECT>/clang/lib/Index/IndexingContext.cpp:84:1
#22 0x00005cfba9fa943a clang::index::IndexingContext::handleDecl(clang::Decl const*, unsigned int, llvm::ArrayRef<clang::index::SymbolRelation>) <LLVM_PROJECT>/clang/lib/Index/IndexingContext.cpp:68:1
#23 0x00005cfba9fad308 VisitVarDecl <LLVM_PROJECT>/clang/lib/Index/IndexDecl.cpp:308:5
#24 0x00005cfba9fad308 VisitDecompositionDecl <CLANGD_BUILD>/tools/clang/include/clang/AST/DeclNodes.inc:332:1
#25 0x00005cfba9fad308 VisitDecompositionDecl <LLVM_PROJECT>/clang/lib/Index/IndexDecl.cpp:317:40
#26 0x00005cfba9fad308 clang::declvisitor::Base<llvm::make_const_ptr, (anonymous namespace)::IndexingDeclVisitor, bool>::Visit(clang::Decl const*) <CLANGD_BUILD>/tools/clang/include/clang/AST/DeclNodes.inc:332:1
#27 0x00005cfba9faea6b clang::index::IndexingContext::indexDecl(clang::Decl const*) <LLVM_PROJECT>/clang/lib/Index/IndexDecl.cpp:789:3
#28 0x00005cfba9faec0c clang::index::IndexingContext::indexDeclContext(clang::DeclContext const*) <LLVM_PROJECT>/clang/lib/Index/IndexDecl.cpp:800:5
#29 0x00005cfba9fad4a0 llvm::SmallVectorTemplateCommon<clang::index::SymbolRelation, void>::isSmall() const <LLVM_PROJECT>/llvm/include/llvm/ADT/SmallVector.h:148:39
#30 0x00005cfba9fad4a0 llvm::SmallVectorImpl<clang::index::SymbolRelation>::~SmallVectorImpl() <LLVM_PROJECT>/llvm/include/llvm/ADT/SmallVector.h:609:23
#31 0x00005cfba9fad4a0 llvm::SmallVector<clang::index::SymbolRelation, 4u>::~SmallVector() <LLVM_PROJECT>/llvm/include/llvm/ADT/SmallVector.h:1230:3
#32 0x00005cfba9fad4a0 VisitTagDecl <LLVM_PROJECT>/clang/lib/Index/IndexDecl.cpp:371:7
#33 0x00005cfba9fad4a0 VisitRecordDecl <CLANGD_BUILD>/tools/clang/include/clang/AST/DeclNodes.inc:428:1
#34 0x00005cfba9fad4a0 VisitCXXRecordDecl <CLANGD_BUILD>/tools/clang/include/clang/AST/DeclNodes.inc:432:1
#35 0x00005cfba9fad4a0 clang::declvisitor::Base<llvm::make_const_ptr, (anonymous namespace)::IndexingDeclVisitor, bool>::Visit(clang::Decl const*) <CLANGD_BUILD>/tools/clang/include/clang/AST/DeclNodes.inc:432:1
#36 0x00005cfba9faea6b clang::index::IndexingContext::indexDecl(clang::Decl const*) <LLVM_PROJECT>/clang/lib/Index/IndexDecl.cpp:789:3
#37 0x00005cfba9faddc0 clang::index::IndexingContext::indexDeclContext(clang::DeclContext const*) <LLVM_PROJECT>/clang/lib/Index/IndexDecl.cpp:800:5
#38 0x00005cfba9faddc0 VisitNamespaceDecl <LLVM_PROJECT>/clang/lib/Index/IndexDecl.cpp:594:30
#39 0x00005cfba9faddc0 clang::declvisitor::Base<llvm::make_const_ptr, (anonymous namespace)::IndexingDeclVisitor, bool>::Visit(clang::Decl const*) <CLANGD_BUILD>/tools/clang/include/clang/AST/DeclNodes.inc:584:1
#40 0x00005cfba9faea6b clang::index::IndexingContext::indexDecl(clang::Decl const*) <LLVM_PROJECT>/clang/lib/Index/IndexDecl.cpp:789:3
#41 0x00005cfba9fa789c memcpy /usr/include/x86_64-linux-gnu/bits/string_fortified.h:29:33
#42 0x00005cfba9fa789c llvm::detail::PunnedPointer<clang::Decl*>::asInt() const <LLVM_PROJECT>/llvm/include/llvm/ADT/PointerIntPair.h:41:16
#43 0x00005cfba9fa789c llvm::detail::PunnedPointer<clang::Decl*>::operator long() const <LLVM_PROJECT>/llvm/include/llvm/ADT/PointerIntPair.h:45:53
#44 0x00005cfba9fa789c llvm::PointerIntPair<clang::Decl*, 3u, clang::Decl::ModuleOwnershipKind, llvm::PointerLikeTypeTraits<clang::Decl*>, llvm::PointerIntPairInfo<clang::Decl*, 3u, llvm::PointerLikeTypeTraits<clang::Decl*>>>::getPointer() const <LLVM_PROJECT>/llvm/include/llvm/ADT/PointerIntPair.h:94:57
#45 0x00005cfba9fa789c clang::Decl::getNextDeclInContext() <LLVM_PROJECT>/clang/include/clang/AST/DeclBase.h:453:72
#46 0x00005cfba9fa789c clang::DeclContext::decl_iterator::operator++() <LLVM_PROJECT>/clang/include/clang/AST/DeclBase.h:2363:46
#47 0x00005cfba9fa789c (anonymous namespace)::IndexASTConsumer::HandleTranslationUnit(clang::ASTContext&) <LLVM_PROJECT>/clang/lib/Index/IndexingAction.cpp:147:59
#48 0x00005cfbaa542d3c clang::ParseAST(clang::Sema&, bool, bool) <LLVM_PROJECT>/clang/lib/Parse/ParseAST.cpp:183:34
#49 0x00005cfbaa33c6e7 clang::FrontendAction::Execute() <LLVM_PROJECT>/clang/lib/Frontend/FrontendAction.cpp:1346:38
#50 0x00005cfba9efdaf5 llvm::Error::getPtr() const <LLVM_PROJECT>/llvm/include/llvm/Support/Error.h:278:51
#51 0x00005cfba9efdaf5 llvm::Error::operator bool() <LLVM_PROJECT>/llvm/include/llvm/Support/Error.h:241:22
#52 0x00005cfba9efdaf5 clang::clangd::BackgroundIndex::index(clang::tooling::CompileCommand) <LLVM_PROJECT>/clang-tools-extra/clangd/index/Background.cpp:322:42
#53 0x00005cfba9efea83 std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>::~basic_string() /usr/include/c++/13/bits/basic_string.h:804:19
#54 0x00005cfba9efea83 clang::tooling::CompileCommand::~CompileCommand() <LLVM_PROJECT>/clang/include/clang/Tooling/CompilationDatabase.h:44:8
#55 0x00005cfba9efea83 clang::clangd::BackgroundIndex::indexFileTask(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>)::'lambda'()::operator()() const <LLVM_PROJECT>/clang-tools-extra/clangd/index/Background.cpp:168:43
#56 0x00005cfba9f0974e std::_Optional_base_impl<clang::clangd::BackgroundQueue::Task, std::_Optional_base<clang::clangd::BackgroundQueue::Task, false, false>>::_M_get() /usr/include/c++/13/optional:479:2
#57 0x00005cfba9f0974e std::optional<clang::clangd::BackgroundQueue::Task>::operator->() /usr/include/c++/13/optional:968:45
#58 0x00005cfba9f0974e clang::clangd::BackgroundQueue::work(std::function<void ()>) <LLVM_PROJECT>/clang-tools-extra/clangd/index/BackgroundQueue.cpp:44:13
#59 0x00005cfba9ef928b std::_Function_base::~_Function_base() /usr/include/c++/13/bits/std_function.h:243:11
#60 0x00005cfba9ef928b std::function<void ()>::~function() /usr/include/c++/13/bits/std_function.h:334:11
#61 0x00005cfba9ef928b operator() <LLVM_PROJECT>/clang-tools-extra/clangd/index/Background.cpp:114:37
#62 0x00005cfba9ef928b void llvm::detail::UniqueFunctionBase<void>::CallImpl<clang::clangd::BackgroundIndex::BackgroundIndex(clang::clangd::ThreadsafeFS const&, clang::clangd::GlobalCompilationDatabase const&, llvm::unique_function<clang::clangd::BackgroundIndexStorage* (llvm::StringRef)>, clang::clangd::BackgroundIndex::Options)::'lambda'()>(void*) <LLVM_PROJECT>/llvm/include/llvm/ADT/FunctionExtras.h:212:16
#63 0x00005cfba9f9fb6e llvm::detail::UniqueFunctionBase<void>::operator=(llvm::detail::UniqueFunctionBase<void>&&) <LLVM_PROJECT>/llvm/include/llvm/ADT/FunctionExtras.h:329:5
#64 0x00005cfba9f9fb6e llvm::detail::UniqueFunctionBase<void>::operator=(llvm::detail::UniqueFunctionBase<void>&&) <LLVM_PROJECT>/llvm/include/llvm/ADT/FunctionExtras.h:321:23
#65 0x00005cfba9f9fb6e llvm::unique_function<void ()>::operator=(llvm::unique_function<void ()>&&) <LLVM_PROJECT>/llvm/include/llvm/ADT/FunctionExtras.h:352:20
#66 0x00005cfba9f9fb6e operator() <LLVM_PROJECT>/clang-tools-extra/clangd/support/Threading.cpp:103:14
#67 0x00005cfba9f9fb6e operator()<clang::clangd::AsyncTaskRunner::runAsync(const llvm::Twine&, llvm::unique_function<void()>)::<lambda()>&> <LLVM_PROJECT>/llvm/include/llvm/Support/thread.h:46:16
#68 0x00005cfba9f9fb6e __invoke_impl<void, llvm::thread::GenericThreadProxy<std::tuple<clang::clangd::AsyncTaskRunner::runAsync(const llvm::Twine&, llvm::unique_function<void()>)::<lambda()> > >(void*)::<lambda(auto:5&&, auto:6&& ...)>, clang::clangd::AsyncTaskRunner::runAsync(const llvm::Twine&, llvm::unique_function<void()>)::<lambda()>&> /usr/include/c++/13/bits/invoke.h:61:36
#69 0x00005cfba9f9fb6e __invoke<llvm::thread::GenericThreadProxy<std::tuple<clang::clangd::AsyncTaskRunner::runAsync(const llvm::Twine&, llvm::unique_function<void()>)::<lambda()> > >(void*)::<lambda(auto:5&&, auto:6&& ...)>, clang::clangd::AsyncTaskRunner::runAsync(const llvm::Twine&, llvm::unique_function<void()>)::<lambda()>&> /usr/include/c++/13/bits/invoke.h:96:40
#70 0x00005cfba9f9fb6e __apply_impl<llvm::thread::GenericThreadProxy<std::tuple<clang::clangd::AsyncTaskRunner::runAsync(const llvm::Twine&, llvm::unique_function<void()>)::<lambda()> > >(void*)::<lambda(auto:5&&, auto:6&& ...)>, std::tuple<clang::clangd::AsyncTaskRunner::runAsync(const llvm::Twine&, llvm::unique_function<void()>)::<lambda()> >&, 0> /usr/include/c++/13/tuple:2302:27
#71 0x00005cfba9f9fb6e apply<llvm::thread::GenericThreadProxy<std::tuple<clang::clangd::AsyncTaskRunner::runAsync(const llvm::Twine&, llvm::unique_function<void()>)::<lambda()> > >(void*)::<lambda(auto:5&&, auto:6&& ...)>, std::tuple<clang::clangd::AsyncTaskRunner::runAsync(const llvm::Twine&, llvm::unique_function<void()>)::<lambda()> >&> /usr/include/c++/13/tuple:2313:31
#72 0x00005cfba9f9fb6e GenericThreadProxy<std::tuple<clang::clangd::AsyncTaskRunner::runAsync(const llvm::Twine&, llvm::unique_function<void()>)::<lambda()> > > <LLVM_PROJECT>/llvm/include/llvm/Support/thread.h:44:15
#73 0x00005cfba9f9fb6e void* llvm::thread::ThreadProxy<std::tuple<clang::clangd::AsyncTaskRunner::runAsync(llvm::Twine const&, llvm::unique_function<void ()>)::'lambda0'()>>(void*) <LLVM_PROJECT>/llvm/include/llvm/Support/thread.h:62:36
#74 0x000071b0d589caa4 start_thread ./nptl/pthread_create.c:447:8
#75 0x000071b0d5929c6c clone3 ./misc/../sysdeps/unix/sysv/linux/x86_64/clone3.S:80:0
Exception ignored in: <_io.TextIOWrapper name='<stdout>' mode='w' encoding='utf-8'>
BrokenPipeError: [Errno 32] Broken pipe
<REPO_ROOT>/harness/scripts/run_clangd_didopen_workspace.sh: line 111:    33 Exit 120                { emit_lsp "$INIT"; emit_lsp "$INITIALIZED"; emit_lsp "$DIDOPEN"; sleep "$DIDOPEN_WAIT_SECS"; if [[ -n "$POST_OPEN" ]]; then
    emit_lsp "$POST_OPEN";
fi; emit_lsp "$SHUTDOWN"; emit_lsp "$EXIT"; }
        34 Aborted                 (core dumped) | ( cd "$WORKDIR"; "$TARGET" "${TARGET_ARGS[@]}" )
```

**Comment by HighCommander4** [comment]
The comment for the assertion is:

```c++
    // Sanity check that the comment does not come from the PCH. We choose to
    // not write them into PCH, because they are racy and slow to load.
    assert(!Ctx.getSourceManager().isLoadedSourceLocation(RC->getBeginLoc()));
```

Clangd's intention is to not record comments in the PCH **for the preamble**, but I guess Modules involve using PCH in additional ways that may make this assumption not hold.
